src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final int side;
+
+    public Square(int side, String color) {
+        this.side = side;
+        setColor(color);
+        obtainTheArea();
```

Comment:
Calling `obtainTheArea()` in the constructor is not necessary and may lead to potential issues if the method is overridden in a subclass. Consider removing this method call from the constructor. Instead, you can call `setArea(side * side)` directly within the constructor or rely on the method being called elsewhere in the program flow.